workflow/Snakefile

```diff
@@ -0,0 +1,229 @@
+all_biosamples = list(biosamples_config.index.values)
+
+## get TSS and genefile names for each biosample 
+TSS_files = []
+gene_files = []
+for x in all_biosamples:
+	tss_file = config['params_candidate']['genome_tss']
+	gene_file = config['params_neighborhoods']['genes']
+	if biosamples_config.loc[x, "TSS"] != "None":
+		tss_file = biosamples_config.loc[x, 'TSS']
+	if biosamples_config.loc[x, "genes"] != "None":
+		gene_file = biosamples_config.loc[x, 'genes']
+	TSS_files.append(tss_file)
+	gene_files.append(gene_file)
+				
+biosamples_config["TSS"] = TSS_files
+biosamples_config["genes"] = gene_files
+
+
+RESULTS_DIR = config['predictions_results_dir']
+
+def get_accessibility_file(wildcards):
+	default_accessibility_feature = biosamples_config.loc[wildcards.biosample, 'default_accessibility_feature']
+	return biosamples_config.loc[wildcards.biosample, default_accessibility_feature]
+
+## overall rule to run pipeline
+rule all:
+	input:
+		narrowPeaks_sorted = expand(os.path.join(RESULTS_DIR, "{biosample}", "Peaks", "macs2_peaks.narrowPeak.sorted"), biosample=all_biosamples),
+		candidateRegions = expand(os.path.join(RESULTS_DIR, "{biosample}", "Peaks", "macs2_peaks.narrowPeak.sorted.candidateRegions.bed"),
+	    biosample=all_biosamples),
+		enhancerLists = expand(os.path.join(RESULTS_DIR, "{biosample}", "Neighborhoods", "EnhancerList.txt"),  biosample=all_biosamples),
+		allPutative = expand(os.path.join(RESULTS_DIR, "{biosample}", "Predictions", "EnhancerPredictionsAllPutative.txt.gz"), biosample=all_biosamples),
+		enhPredictions = expand(os.path.join(RESULTS_DIR, "{biosample}", "Predictions", "EnhancerPredictions.csv"), biosample=all_biosamples),
+		allPred = os.path.join(RESULTS_DIR,"AllPredictions.txt.gz"),
+		altTSS = TSS_files,
+		metricsDirectory = expand(os.path.join(RESULTS_DIR, "{biosample}", "Metrics", "QCSummary.p"), biosample=all_biosamples)
+
+	
+## call macs2 -- if multiple accessibility inputs for one biosample, will aggregate into one output
+rule call_macs_peaks: 
+	input:
+		accessibility = get_accessibility_file
+	params:
+		pval = config['params_macs']['pval'],
+		out_dir = config["predictions_results_dir"]
+	conda:
+		"envs/abcenv.yml"
+	output: 
+		narrowPeak = os.path.join(RESULTS_DIR, "{biosample}", "Peaks", "macs2_peaks.narrowPeak")
+	shell: 
+		""" 
+		macs2 callpeak -f AUTO -g hs -p {params.pval} -n macs2 --call-summits --outdir {params.out_dir}/{wildcards.biosample}/Peaks -t {input.accessibility}
+		"""
+
+## sort narrowPeaks
+rule sort_narrowpeaks:
+	input:
+		narrowPeak = os.path.join(RESULTS_DIR, "{biosample}", "Peaks", "macs2_peaks.narrowPeak")
+	params:
+		chrom_sizes = config['params_candidate']['chrom_sizes']
+	conda:
+		"envs/abcenv.yml"
+	output:
+		narrowPeakSorted = os.path.join(RESULTS_DIR, "{biosample}", "Peaks", "macs2_peaks.narrowPeak.sorted")
+	shell:
+		"""
+		grep -v -E "^\w+_.*" {input.narrowPeak} | bedtools sort -faidx {params.chrom_sizes} -i stdin > {output.narrowPeakSorted}
```

Comment:
Note for future: 

The 'grep' step appears to be filtering out non-standard chromosome names e.g. chr1_random.  This step is probably important because some input files will include reads to nonstandard chromosomes that we eventually do want to filter out. But, this may or may not be the right place to do this step.  At very least, this should be more clearly documented and be explicitly controlled by providing an accompanying 'standard' chromosome file that user can specify to do this filter explicitly, e.g. using a bedtools intersect command.

Reply:
ah yes. I changed this to use bedtools intersect in the most updated `dev` branch: https://github.com/broadinstitute/ABC-Enhancer-Gene-Prediction/blob/dev/workflow/Snakefile#L81

Re: documentation. I can add that to ReadTheDocs? Unless you think there's a better place for it